work_queue/src/work_queue_coprocess.c

```diff
@@ -98,6 +118,10 @@ char *work_queue_coprocess_setup(int *coprocess_port)
 		if (curr_bytes_read < 0) {
 			fatal("Unable to get information from coprocess\n");
 		}
+		if (curr_bytes_read == 0)
```

Comment:
Could `4096 - buffer_offset` (e.g. the len argument for w_q_coprocess_{read,write}) ever be zero? If so, I would make it so that `curr_bytes_read` returns -1 when a retry is needed, and > -1 on failure.  Otherwise if 0 is the correct number of bytes read, this may fail